gcovr/html_generator.py

```diff
@@ -373,6 +376,7 @@ def _make_short_sourcename(output_file, filename):
         # filename if it exceeds common filesystem limitations
         if len(os.path.basename(sourcename)) < 256:
```

Comment:
The limit is 260 characters (MAX_PATH).

Reply:
Thank you for pointing.
Maybe it's needed to consider all path and not only sourcename.
What do you think?

Reply:
All paths should respect this limit. What about using only the basename of the file extended by the md5sum of the whole path e.g.:
`path/to/source.c` --> `source.c.883f9ee88c14d9c77addd640015ee77d.html`

Reply:
There is already that:
```python
        longname_hash = "_" + hex(zlib.crc32(longname.encode('utf-8')) & 0xffffffff)[2:]
        longname = longname[(len(sourcename) - len(longname_hash)):]
```
Isn't good enough?

Reply:
I suggested to always use the naming schema instead of removing the known bad characters and use a hash if the name gets to long.

Reply:
Sorry I didn't undestand that before.
I'm agree with you, i think is the best approach to avoid path errors.

Would you like me to do it?

Reply:
I think @latk should decide this. I also prefer md5 instead of crc32.